langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -0,0 +1,426 @@
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static dev.langchain4j.internal.Utils.*;
+import static dev.langchain4j.internal.Utils.isNullOrEmpty;
+import static dev.langchain4j.internal.ValidationUtils.*;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Azure AI Search EmbeddingStore Implementation
+ */
+public class AzureAiSearchEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+    private static final Logger log = LoggerFactory.getLogger(AzureAiSearchEmbeddingStore.class);
+
+    private static final String INDEX_NAME = "vectorsearch";
+
+    private static final String DEFAULT_FIELD_ID = "id";
+
+    private static final String DEFAULT_FIELD_CONTENT = "content";
+
+    private static final String DEFAULT_FIELD_CONTENT_VECTOR = "content_vector";
+
+    private static final String DEFAULT_FIELD_METADATA = "metadata";
+
+    private static final String DEFAULT_FIELD_METADATA_SOURCE = "source";
+
+    private static final String DEFAULT_FIELD_METADATA_ATTRS = "attributes";
+
+    private final SearchIndexClient searchIndexClient;
+
+    private final SearchClient searchClient;
+
+    public AzureAiSearchEmbeddingStore(String endpoint, AzureKeyCredential keyCredential, EmbeddingModel embeddingModel) {
+
+        searchIndexClient = new SearchIndexClientBuilder()
+                .endpoint(endpoint)
+                .credential(keyCredential)
+                .buildClient();
+
+        searchClient = new SearchClientBuilder()
+                .endpoint(endpoint)
+                .credential(keyCredential)
+                .indexName(INDEX_NAME)
+                .buildClient();
+
+        createOrUpdateIndex(embeddingModel);
+    }
+
+    public AzureAiSearchEmbeddingStore(String endpoint, TokenCredential tokenCredential, EmbeddingModel embeddingModel) {
+
+        searchIndexClient = new SearchIndexClientBuilder()
+                .endpoint(endpoint)
+                .credential(tokenCredential)
+                .buildClient();
+
+        searchClient = new SearchClientBuilder()
+                .endpoint(endpoint)
+                .credential(tokenCredential)
+                .indexName(INDEX_NAME)
+                .buildClient();
+
+        createOrUpdateIndex(embeddingModel);
+    }
+
+     void createOrUpdateIndex(EmbeddingModel embeddingModel) {
+
+        // Embed a test query to get the embedding dimensions
+        int embeddingDimensions = embeddingModel.embed("test").content().vector().length;
```

Comment:
I would suggest accepting an `int dimensions` parameter into ctor/builder instead.
One should be able to create `EmbeddingStore` without `EmbeddingModel`.

Reply:
Indeed, that's much better!! Fixed.